sw/host/opentitanlib/src/transport/hyperdebug/dfu.rs

```diff
@@ -462,3 +486,55 @@ fn wait_for_idle(dfu_device: &UsbBackend, dfu_interface: u8) -> Result<u8> {
         }
     }
 }
+
+/// Returns true if the two version strings have the same text prefix, e.g. "hyperdebug_",
+/// differing only in the subsequent numbers, and furthermore that the numbers in `version_a` are
+/// strictly "less than" those in `version_b`.
+fn is_older_than(version_a: &str, version_b: &str) -> Result<bool> {
```

Comment:
I could not find references to "numerical sort" being used when comparing tuples of multiple numbers, on the other hand, I have heard ["lexicographical sort"](https://wiki.python.org/moin/HowTo/Sorting/) being used when tuples of multiple number are sorted (the values of each coordinate being numerically compared to their counterparts, of course).  Also, this particular algorithm treats non-comparable values as "not older than", because that is what we need in our particular case.

Because of the above, I have a hard time coming up with a concise way of describing what it does, other than maybe `version_strings_comparable_and_older_than()`, and I do not think it is ready to be put into `util`, until we see maybe one other use case for it, and figure out how to properly declare it in a reusable fashion.